src/DealClient.sol

```diff
@@ -99,6 +99,22 @@ contract DealClient {
         owner = msg.sender;
     }
 
+    function getProviderSet(bytes calldata cid) public view returns (ProviderSet memory){
```

Comment:
Do we really need this getters? mappings should serve as getters for the external world?

Reply:
yes i think we actually really do :(:(

Reply:
i'm looking to confirm or see if there's a work around

Reply:
![image](https://user-images.githubusercontent.com/1981537/222627253-c57450d1-c5f8-40f6-b885-59a81965cdce.png)
we definitely need a getter for .length() - i think it's possible to access structs in the contract from the test contract without a getter, but it's much more annoying and you need to unpack the struct etc. the getters get the job done. I also think other people could want these getters so it's a win win

Reply:
hmm this is annoying
I'd prefer not to accumulate a lot of these getters in the contract since these structures are never going to be used externally. They're only purpose will be to add a test for the implementation detail

I believe something like this works
`(bytes32 id, bool valid) = client.pieceToProposal(testCID);
  require(valid, "expected to have");`